scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -890,10 +891,14 @@ def visit_functiondef(self, node):
                     try:
                         # infer_call_result gives the method return value as a string
                         returns = next(node.infer_call_result()).as_string()
-                        if returns.find("ItemPaged") == -1 and returns.find("AsyncItemPaged") == -1:
-                            self.add_message(
-                                msgid="client-list-methods-use-paging", node=node, confidence=None
-                            )
+  
+                        # Also check the docstring return type and make sure rtype is there
+                        if node.doc.split("rtype:")[-1] not in returns:
```

Comment:
Not 100% on this,  but won't `node.doc.split("rtype:")[-1]` just return everything in the docstring after `rtype`? Which might include sample snippet admonitions and versionadded directives?